### PR TITLE
[OM] Add evaluator support for path append

### PR DESCRIPTION
**Pull request by prithayan**
Evaluate the path append op.

### CODE REVIEW

File: lib/Dialect/OM/Evaluator/Evaluator.cpp [hunk 1]
```diff
@@ -115,6 +116,37 @@ circt::om::Evaluator::instantiate(
   return success(std::shared_ptr<evaluator::ObjectValue>(object));
 }
 
+bool hasOMTarget(StringRef path) {
+  return (path.find("OMDontTouchedReferenceTarget") == 0 ||
```
Comment (uenoku) [inline lib/Dialect/OM/Evaluator/Evaluator.cpp:120]: path.startwith would be nicer. Either way did we consider adding target kind to a path attribute as a field?

File: lib/Dialect/OM/Evaluator/Evaluator.cpp [hunk 2]
```diff
@@ -148,6 +180,32 @@ FailureOr<evaluator::EvaluatorValuePtr> circt::om::Evaluator::evaluateValue(
...
+              auto suffixVal = evaluateValue(op.getPath(), actualParams);
+              auto rootPath = cast<evaluator::AttributeValue>(rootVal->get())
+                                  ->getAttr()
+                                  .cast<om::PathAttr>()
+                                  .getPath()
+                                  .getValue();
+              auto suffixPath =
+                  cast<evaluator::AttributeValue>(suffixVal->get())
+                      ->getAttr()
+                      .cast<om::PathAttr>()
```
Comment (uenoku) [inline lib/Dialect/OM/Evaluator/Evaluator.cpp:195]: nit:
```suggestion
->getAs<om::PathAttr>
```

File: lib/Dialect/OM/Evaluator/Evaluator.cpp [hunk 3]
```diff
@@ -120,6 +121,37 @@ circt::om::Evaluator::instantiate(
   return success(std::shared_ptr<evaluator::ObjectValue>(object));
 }
 
+bool hasOMTarget(StringRef path) {
+  return (path.startswith("OMDontTouchedReferenceTarget") ||
+          path.startswith("OMMemberInstanceTarget") ||
+          path.startswith("OMMemberReferenceTarget") ||
+          path.startswith("OMReferenceTarget"));
```
Comment (dtzSiFive) [inline lib/Dialect/OM/Evaluator/Evaluator.cpp:128]: + `OMInstanceTarget` (Chisel emits, added to CIRCT after this PR I believe).

This already exists, but gosh it'd be nice if every pass touching this concept didn't have its own local list of the "recognized" target kinds.

File: lib/Dialect/OM/Evaluator/Evaluator.cpp [hunk 4]
```diff
@@ -120,6 +121,37 @@ circt::om::Evaluator::instantiate(
...
+}
+
+/// Parse origTarget and prepend the root path.
+std::string addPathPrefix(StringRef origTarget, StringRef rootPath) {
```
Comment (dtzSiFive) [inline lib/Dialect/OM/Evaluator/Evaluator.cpp:132]: I was going to suggest using FIRRTL's `tokenizePath` but I see there are some differences.

Once we pin down how this works (if haven't already), may be good to have this have error paths if things aren't as expected / invalid formats?

**Review by dtzSiFive — COMMENTED** [review]
Drive-by comments, mostly to note the new path kind needs to be added.

**Comment by prithayan** [comment]
No longer relevant.

**Pull request closed without merging**